pkg/file/io.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package file
+
+import (
+	"bytes"
+	"errors"
+
+	"io"
+)
+
+// simpleJoinerReadCloser wraps a byte slice in a io.ReadCloser implementation.
+type simpleReadCloser struct {
+	buffer io.Reader
+	closed bool
+}
+
+func NewSimpleReadCloser(buffer []byte) io.ReadCloser {
+	return &simpleReadCloser{
+		buffer: bytes.NewBuffer(buffer),
+	}
+}
+
+// Read implements io.Reader.
+func (s *simpleReadCloser) Read(b []byte) (int, error) {
+	if s.closed {
```

Comment:
race detector might show this as a possible race, since there's no mutex to protect the field. fyi

Reply:
consider using [Value](https://golang.org/pkg/sync/atomic/#Value)

Reply:
k